### ISSUE TITLE
ssl_client2 example cannot send request of length 0

### ISSUE THREAD

**Issue by andresag01** [ARMmbed/mbedtls#1833]
### Description
- Type: Bug
- Priority: Minor

---------------------------------------------------------------
## Bug

The example at programs/ssl/ssl_client2.c states the following in the help message:
```
    request_size=%d     default: about 34 (basic request)
                        (minimum: 0, max: 20000 )
```
However, in practice the application fails to send 0-length application data messages because of the following code:
```
        for( written = 0, frags = 0; written < len; written += ret, frags++ )
        {
            while( ( ret = mbedtls_ssl_write( &ssl, buf + written,
                                              len - written ) ) <= 0 )
            {
```
What currently happens is that the message is never actually sent and the ssl_client2 directly moves on to wait for an incoming message from the server to arrive. In fact, the following commands result in a deadlock:
```
./programs/ssl/ssl_server2 server_port=3344 server_addr=127.0.0.1 force_version=tls1 arc4=1 ca_file=none auth_mode=none crt_file=data_files/server2.crt key_file=data_files/server2.key allow_sha1=1 debug_level=4 etm=1 &
./programs/ssl/ssl_client2 server_port=3344 server_addr=127.0.0.1 force_version=tls1 arc4=1 ca_file=none auth_mode=none allow_sha1=1 debug_level=4 etm=1 force_ciphersuite=TLS-ECDHE-RSA-WITH-AES-256-CBC-SHA request_size=0
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-2390

**Comment by RonEld** [comment]
issue was fixed by #1852 which was merged by now, therefore closing
